### PR TITLE
TileRequest handles 404 errors thrown by @itwin/object-storage

### LINKED ISSUE #4532: Tiles are not requested from backend after blob storage request fails
### Discussed in https://github.com/iTwin/itwinjs-core/discussions/4531

@aruniverse @calebmshafer @paulius-valiunas @wgoehrig @swbsi @nick4598
`IModelTileRpcInterface.requestTileTreeProps` succeeds and returns tile trees. We then attempt and fail to download tiles from blob storage, as expected. No subsequent requests are made to `IModelTileRpcInterface` to generate the tiles.

In [TileRequest.dispatch](https://github.com/iTwin/itwinjs-core/blob/e9cda85a63437b1d09644fed8432d2b90ff9ee35/core/frontend/src/tile/TileRequest.ts#L89) an error is caught:
```
"RestError: The specified blob does not exist.
RequestId:82eb8eb2-601e-0035-3a34-e640d4000000
Time:2022-10-22T16:39:51.4561057Z
    at new RestError (http://localhost:3000/static/js/vendors~object-storage.chunk.js:3418:28)
    at handleErrorResponse (http://localhost:3000/static/js/vendors~object-storage.chunk.js:1962:17)
    at http://localhost:3000/static/js/vendors~object-storage.chunk.js:1898:18"
```

Previously, I believe `requestContent` would have returned `undefined` instead of throwing an error, and we would have proceeded via [onNoContent](https://github.com/iTwin/itwinjs-core/blob/e9cda85a63437b1d09644fed8432d2b90ff9ee35/core/frontend/src/tile/IModelTileRequestChannels.ts#L25) to request the tile via rpc.
I recall some changes were made to HTTP status handling in 3.4 which may be relevant.
We need to figure out how this went undiscovered until just now.

### PR DESCRIPTION
**Pull request by pmconne**
Fixes #4532.

### CODE REVIEW

**Comment by pmconne** [comment]
@Mergifyio backport release/3.4.x

File: core/frontend/src/tile/TileRequest.ts [hunk 1]
```diff
@@ -92,7 +92,12 @@ export class TileRequest {
       // Set this now, so our `isCanceled` check can see it.
       this._state = TileRequest.State.Loading;
     } catch (err: any) {
-      if (err.errorNumber && err.errorNumber === IModelStatus.ServerTimeout) {
+      if (err.statusCode === 404) {
```
[the remark this answers is not in the document]
Comment (pmconne) [reply core/frontend/src/tile/TileRequest.ts:95]: That sounds wrong, for a package that claims to expose a "unified cloud-agnostic object storage interface".

This is an issue in production right now. We need to fix it now.

[the remark this answers is not in the document]
Comment (paulius-valiunas) [reply core/frontend/src/tile/TileRequest.ts:95]: Well I think if a request to the tile cache fails, it's OK to treat it as a cache miss at least for now. It's not ideal, but at least it works most of the time and won't break anything even if a cache miss is registered for a different error.

That said, I have an alternative solution at #4535, which reduces the complexity a bit by removing `undefined` from the equation and moves cloud cache-specific error handling out of `TileRequest`.

As for error handling at object-storage, I agree it's wrong, and I don't think we can expect to have a reliable way to detect 404 errors until they implement proper error handling on their side.

[the remark this answers is not in the document]
Comment (pmconne) [reply core/frontend/src/tile/TileRequest.ts]: Your PR changes too much in too many places for a temporary fix.
Please re-review this PR.

**Review by paulius-valiunas — APPROVED** [review]
Looks good as a temporary fix. We should revisit this later to make it less messy, preferably involving an update to object-storage for the error handling.

**Pull request merged**